google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpannerOptionsTest.java

```diff
@@ -402,4 +404,19 @@ public void testSetClientLibToken() {
   public void testSetInvalidClientLibToken() {
     SpannerOptions.newBuilder().setClientLibToken("foo");
   }
+
+  @Test
+  public void testSetEmulatorHostEnv() {
+    environmentVariables.set("SPANNER_EMULATOR_HOST", "localhost:1234");
+    SpannerOptions options = SpannerOptions.newBuilder().setProjectId("[PROJECT]").build();
+    assertThat(options.getHost()).isEqualTo("http://localhost:1234");
+    assertThat(options.getEndpoint()).isEqualTo("localhost:1234");
+
+    environmentVariables.set("SPANNER_EMULATOR_HOST", "http://localhost:1234");
```

Comment:
This looks like you're testing the same thing twice.

Reply:
Originally, the first case would throw an error when converting it to a Url object because of lack of protocol (in getEndpoint() method).

Reply:
Could this be two methods/ I still don't follow what's going on here. A comment m,ight be helpful.

Reply:
The current convention is to assign `localhost:1234` to `SPANNER_EMULATOR_HOST` without the protocol prefix.